.github/workflows/deploy.yml

```diff
@@ -25,10 +25,16 @@ jobs:
 
     - name: Build
       run: |
-        jupyter-book build .
+        jupyter-book build -W -n --keep-going .
 
     - name: Upload
-      if: github.event_name != 'pull_request'
+      if: |
+        (
+          github.event_name != 'pull_request' && (
+            github.ref_name != 'main' ||
+            github.event_name == 'workflow_dispatch'
+          )
+        )
```

Comment:
Prevents publishing to pages on a PR, a branch that is not main or a manual run.